tests/integration.rs

```diff
@@ -195,19 +202,133 @@ fn test_allow_any_unexpected_error_status() {
 fn test_whitelist_unexpected_error_status() {
     //! A response from an error inside a handler should contain CORS headers.
     let mut handler = Chain::new(ErrorResultHandler {});
-    let whitelist = ["example.org"].iter().map(ToString::to_string).collect::<HashSet<_>>();
+    let whitelist = ["http://example.org:3000"].iter().map(ToString::to_string).collect::<HashSet<_>>();
     handler.link_around(CorsMiddleware::with_whitelist(whitelist));
-    let headers = setup_origin_header!("example.org");
+    let headers = setup_origin_header!("example.org", 3000);
     let error = request::get("http://example.org:3000/err", headers, &handler).unwrap_err();
     let response = error.response;
     assert_eq!(response.status, Some(status::InternalServerError));
 
     {
     let header = response.headers.get::<AccessControlAllowOrigin>();
     assert!(header.is_some());
-    assert_eq!(*header.unwrap(), AccessControlAllowOrigin::Value("http://example.org".into()));
+    assert_eq!(*header.unwrap(), AccessControlAllowOrigin::Value("http://example.org:3000".into()));
     }
 
     let result_body = response::extract_body_to_string(response);
     assert_eq!(&result_body, "Oh noes");
 }
+
+#[test]
+fn test_whitelist_preflight_with_cors_headers() {
+    //! OPTION requests with whitelisted host and correct CORS headers should answer 200 with empty body and the CORS headers 
+    let handler = setup_handler!("whitelist": ["http://example.org:3000"]);
+    
+    let headers = {
+        let mut headers = Headers::new();
+        headers.set(Origin::new("http", "example.org", Some(3000)));
+        headers.set(AccessControlRequestMethod(iron::method::Get));
+        headers.set(AccessControlRequestHeaders(vec!(UniCase("header1".to_string()),UniCase("header2".to_string()))));
```

Comment:
Yes! I didn't realize and I wrote that with () instead of []. 
https://users.rust-lang.org/t/braces-square-brackets-and-parentheses-in-macro-call/9984
I will change once I arrive home.